projects/element-ng/chat-messages/chat-message.model.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) Siemens 2016 - 2025
+ * SPDX-License-Identifier: MIT
+ */
+import type { Signal, TemplateRef } from '@angular/core';
+import type { TranslatableString } from '@siemens/element-translate-ng/translate-types';
+
+/**
+ * Actions for messages representing an action with icon, label (for accessibility), and handler.
+ * Only the icon will be displayed.
+ * @experimental
+ */
+export interface MessageAction {
+  /** Label that is shown to the user. */
+  label: TranslatableString;
+  /**
+   * Icon used to represent the action
+   */
+  icon: string;
+  /**
+   * Action that is called when the item is triggered.
+   */
+  action: (actionParam: any, source: this) => void;
+  /** Whether the menu item is disabled. */
+  disabled?: boolean;
+}
+
+/**
+ * Attachment item interface for file attachments in chat messages, used by {@link SiAttachmentListComponent} and inside {@link SiUserMessageComponent} as well as {@link SiChatInputComponent}.
+ *
+ * @see {@link SiAttachmentListComponent} for the attachment list component
+ * @see {@link SiUserMessageComponent} for the user message
+ * @see {@link SiChatInputComponent} for the chat input component
+ *
+ * @experimental
+ */
+export interface Attachment {
+  /** File name */
+  name: string;
+  /** Optionally show a preview of the attachment by providing a template that is shown in a modal when clicked (optional) */
+  previewTemplate?: TemplateRef<any> | (() => TemplateRef<any>);
+}
+
+/**
+ * Base interface for all chat messages. Messages can be rendered either with content
+ * or with a custom template for advanced styling and functionality.
+ * @experimental
+ */
+export interface BaseChatMessage {
+  /** Type of message */
+  type: 'user' | 'ai';
+  /** Message content - can be a string or a Signal<string>, empty string shows loading state */
+  content?: string | Signal<string>;
+  /** Whether the message is currently loading/being generated - can be a boolean or Signal<boolean> */
+  loading?: boolean | Signal<boolean>;
+}
+
+/**
+ * User chat message
+ * @experimental
+ */
+export interface UserChatMessage extends BaseChatMessage {
+  /** Type of message */
+  type: 'user';
+  /** Message content, should be a string, empty string shows loading state */
+  content: string;
+  /** Attachments (for user messages) */
+  attachments?: Attachment[];
+  /** Actions available for this message */
+  actions?: MessageAction[];
+}
+
+/**
+ * AI chat message
+ * @experimental
+ */
+export interface AiChatMessage extends BaseChatMessage {
```

Comment:
@Killusions Btw: What happens when the AI responds with pictures or documents?

Reply:
Yes, AI message has loading state and user message can have attachments. But most importantly it's used to decide the styling